modules/ocs2/src/main/scala/gem/ocs2/Parsers.scala

```diff
@@ -215,4 +215,277 @@ object Parsers {
       "OPEN"  -> F2WindowCover.Open
     )
   }
+
+
+  object Gmos {
+    import GmosAdc._
+
+    val adc: PioParse[Option[GmosAdc]] = enum(
+      "No Correction"          -> Option.empty[GmosAdc],
+      "Best Static Correction" -> Some(BestStatic),
+      "Follow During Exposure" -> Some(Follow)
+    )
+
+
+    import GmosAmpCount._
+
+    val ampCount: PioParse[GmosAmpCount] = enum(
+      "Three"  -> Three,
+      "Six"    -> Six,
+      "Twelve" -> Twelve
+    )
+
+
+    import GmosAmpGain._
+
+    val ampGain: PioParse[GmosAmpGain] = enum(
+      "Low"  -> Low,
+      "High" -> High
+    )
+
+
+    import GmosAmpReadMode._
+
+    val ampReadMode: PioParse[GmosAmpReadMode] = enum(
+      "Slow" -> Slow,
+      "Fast" -> Fast
+    )
+
+
+    val binning: PioParse[GmosBinning] = enum(
+      "1" -> GmosBinning.One,
+      "2" -> GmosBinning.Two,
+      "4" -> GmosBinning.Four
+    )
+
+
+    import GmosBuiltinRoi._
+
+    val builtinRoi: PioParse[Option[GmosBuiltinRoi]] = enum(
+      "Custom ROI"         -> Option.empty[GmosBuiltinRoi],
+      "Full Frame Readout" -> Some(FullFrame      ),
+      "CCD 2"              -> Some(Ccd2           ),
+      "Central Spectrum"   -> Some(CentralSpectrum),
+      "Central Stamp"      -> Some(CentralStamp   ),
+      "Top Spectrum"       -> Some(TopSpectrum    ),
+      "Bottom Spectrum"    -> Some(BottomSpectrum ),
+    )
```

Comment:
Interesting, this seems like it shouldn't compile but it does.  It does trigger the scalastyle error though.